mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/JobClusterActor.java

```diff
@@ -1895,6 +1900,37 @@ private long getExpireAcceptedDelayMs() {
         return empty();
     }
 
+
+    private Optional<JobDefinition> createJobDefinitionInheritingInstanceCount(
+            final JobDefinition givenJobDefn,
+            final List<JobInfo> existingJobsList) {
+        if(logger.isTraceEnabled()) { logger.trace("Enter createJobDefinitionInheritingInstanceCount"); }
+
+        // use last submitted job (if non-terminal) to merge with the givenJobDefn so the result jobDefn will inherit instance count.
+        JobDefinition finalJobDefn = givenJobDefn;
+        Optional<JobId> lastJobId = JobListHelper.getLastSubmittedJobId(existingJobsList, Lists.newArrayList());
+        if (lastJobId.isPresent()) {
+            Optional<JobInfo> jobInfoForNonTerminalJob = jobManager.getJobInfoForNonTerminalJob(lastJobId.get());
+            if (jobInfoForNonTerminalJob.isPresent()) {
+                CompletableFuture<Optional<JobDefinition>> mergedJobDefnOCS =
+                        getNonTerminalJobDefnWithWorkerCountInheritance(jobInfoForNonTerminalJob.get(), givenJobDefn, false)
+                                .toCompletableFuture();
+                try {
+                    if (mergedJobDefnOCS.get(500, TimeUnit.MILLISECONDS).isPresent()) {
+                        finalJobDefn = mergedJobDefnOCS.get().get();
+                    }
+                } catch (InterruptedException | ExecutionException | TimeoutException e) {
+                    logger.error("Failed to merge with running job's instance", e);
+                }
+            }
+        }
+        else {
+            logger.info("No running job instance available to inherit stage worker instance {}", givenJobDefn.getName());
+        }
+
+        return createNewJobDefinitionInheritSchedInfoAndParameters(finalJobDefn);
```

Comment:
I think this should just return givenJobDefn. This should behave the same as if inherit count is not set.

Reply:
Renamed and updated the logic.